ui/src/shared/components/TableGraph.tsx

```diff
@@ -310,33 +305,79 @@ class TableGraph extends Component<Props, State> {
         columnWidths: columnWidths.widths,
         totalColumnWidths: columnWidths.totalWidths,
         isTimeVisible,
+        shouldResize: true,
       })
     }
   }
 
+  public componentDidUpdate() {
+    if (this.state.shouldResize) {
+      if (this.multiGrid) {
+        this.multiGrid.recomputeGridSize()
+      }
+
+      this.setState({shouldResize: false})
+    }
+  }
+
+  private hasDataChanged(data): boolean {
+    const newUUID = _.get(data, '0.response.uuid', null)
+    const oldUUID = _.get(this.props.data, '0.response.uuid', null)
+
+    return newUUID !== oldUUID
+  }
+
+  private get fixFirstColumn(): boolean {
+    const {tableOptions} = this.props
+    const {fixFirstColumn = DEFAULT_FIX_FIRST_COLUMN} = tableOptions
+    return fixFirstColumn
+  }
+
+  private get columnCount(): number {
+    const {transformedData} = this.state
+    return _.get(transformedData, ['0', 'length'], 0)
+  }
+
+  private get computedColumnCount(): number {
+    if (this.fixFirstColumn) {
+      return this.columnCount - 1
+    }
+
+    return this.columnCount
+  }
+
+  private get tableWidth(): number {
+    const tableWidth = _.get(this, ['gridContainer', 'clientWidth'], 0)
+
+    return tableWidth
+  }
+
+  private get computedTableWidth(): number {
```

Comment:
do we need a getter that just returns another getter? consider just using `tableWidth`